### ISSUE TITLE
Reporting / Dashboard is including voided orders from totals

### ISSUE THREAD

**Issue by dxdc** [ingenesis/shopp#3262]
I've done some careful analysis, and Shopp is not excluding canceled/voided orders from its totaling in the reporting. So, it includes revenue and product numbers from those orders. I think it might be appropriate to include "order type" here at the very least to have it only total Completed orders for instance.

Also, see my note here regarding modifying this Reporting to be more versatile for accounting. There might be a way to consolidate both of these and make it more powerful.

https://github.com/ingenesis/shopp/issues/3259

**Comment by dxdc** [comment]
I'm wondering if this is because payments are failing, e.g.:

March 5 1:01 pm	Order invoiced		$9.95
March 5 1:01 pm	Payment authorization & capture		
March 5 1:01 pm	Authorization failed	Your card number is incorrect.	
March 5 1:02 pm	Payment authorization & capture		
March 5 1:02 pm	Authorization failed	Your card number is incorrect.	
March 5 1:02 pm	Payment authorization & capture		
March 5 1:02 pm	Authorization failed	Your card number is incorrect.	
March 5 1:04 pm	Payment authorization & capture		
March 5 1:04 pm	Authorization failed	Your card number is incorrect.
Thursday 10:07 pm	Notice	Marked Cancelled by  xx@.net	
Thursday 10:07 pm	Order cancelled

**Comment by Just-m** [comment]
According to the code it is only including three types of txnstatus https://github.com/ingenesis/shopp/blob/391abe3c6588f08a460d762cfae9bb41b51f5559/core/flow/Dashboard.php#L132

**Comment by dxdc** [comment]
Right... I think the issue is that there are authorization entries in my log for those orders.

**Comment by Just-m** [comment]
Can you check the shopp_purchase table?

**Comment by dxdc** [comment]
It says voided there. Strange...?

**Comment by Just-m** [comment]
Yep. Can you echo the $query? Or write it to Shopp log?

**Comment by dxdc** [comment]
What's the easy DEBUG command for that? I'll have to look at it later.

**Comment by Just-m** [comment]
I use `ShoppError('===== query ====='._object_r($query), false, SHOPP_DEBUG_ERR);`

**Comment by dxdc** [comment]
OK thanks.

Hmm, this is so odd. I went back and now it's reporting the correct number of orders. The only thing I can think is that it seems I've only noticed this when "today" is selected from the dropdown. I'll wait until I have another canceled order and can re-evaluate.

I was having some issues with db caching a few weeks back, but I thought I had resolved those.

**Comment by crunnells** [comment]
A little off-topic, but 1.3.x has a great debugging function: `shopp_debug('Some Debugging Message here')` will output whatever text (or variables) to the log only if it's set to Debugging.

**Comment by dxdc** [comment]
thanks for sharing @crunnells ! that is fantastic.

**Comment by dxdc** [comment]
I still haven't had a canceled order to test, but here's one thing I have noticed now after some more thorough troubleshooting.

The Dashboard reporting module reports the wrong values.

If I choose "this week" then "last week" then "this week" then "last week" (cycling through these options), the stats reported are definitely wrong in the best seller list and maybe in the totals too. For instance, there are products in the best seller list that definitely didn't even sell in that time period.

Perhaps this is the crux of the issue that I originally discovered. Also, depending on which item I choose, (today, yesterday, etc.) sometimes that best seller list is completely blank.

**Comment by Just-m** [comment]
Did you update to 1.3.9? Any caching software in use?

**Comment by dxdc** [comment]
Turned off all caching for the test, but haven't upgraded yet.

**Comment by Just-m** [comment]
I am not saying there is nothing wrong with the code, just trying to prevent looking for something that is interfering outside Shopp. Turning off caching does not always result in completely disabling caching.
Did you check your wp-config.php for 'still active' commands?

**Comment by dxdc** [comment]
Thanks.. there's nothing else there though. Do your numbers change if you flip back and forth like I've just described?

**Comment by dxdc** [comment]
Also @Just-m, another oddiity I just noticed:

On the Dashboard/Shopp reporting module, under the "Lifetime" orders, this reports a DIFFERENT NUMBER (slightly lower) than the number of orders that I see if I click on the Shopp > Orders tab.

I had 1.18 installed and have upgraded since, but is this a historic issue with the db or an actual bug that needs to be corrected?
